site/server.ts

```
@@ -50,6 +50,21 @@ app.prepare().then(() => {
                 };
             }
 
+            const originalWriteHead = res.writeHead;
+            res.writeHead = function (statusCode: number, ...args: unknown[]) {
+                const statusCodeClass = getStatusCodeClass(statusCode);
+                if (statusCodeClass === "client_error" || statusCodeClass === "server_error") {
+                    // set no-store, max-age=0 to prevent caching of error responses
+                    res.setHeader("Cache-Control", "no-store, max-age=0");
+                }
+
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-ignore
+                // Reason: TS complains about passing unknown[] to writeHead, however explicit typing is complicated
+                // because writeHead can be overloaded and has two signatures
+                return originalWriteHead.apply(this, [statusCode, ...args]);
```

Passing the args as unknown[] to originalWriteHead causes the typescript error `Type 'unknown' is not assignable to type 'OutgoingHttpHeaders | OutgoingHttpHeader[] | undefined'.`. This only happens in the starter, it didn't happen in Demo.

Since `writeHead` can be overloaded and has two signatures, correct typing is hard. So I opted for just ignoring the error here.